backend/infrahub/core/migrations/graph/patch_0001.py

```diff
@@ -0,0 +1,124 @@
+CALL {
+    WITH node_with_dup_edges, edge_type, edge_branch, peer
+    MATCH (node_with_dup_edges)-[active_edge {branch: edge_branch, status: "active"}]-(peer)
+    WHERE type(active_edge) = edge_type
+    WITH node_with_dup_edges, edge_type, edge_branch, peer, active_edge
+    ORDER BY active_edge.from ASC
+    WITH node_with_dup_edges, edge_type, edge_branch, peer, head(collect(active_edge.from)) AS active_from
+    OPTIONAL MATCH (node_with_dup_edges)-[deleted_edge {branch: edge_branch, status: "deleted"}]->(peer)
+    WITH node_with_dup_edges, edge_type, edge_branch, peer, active_from, deleted_edge
+    ORDER BY deleted_edge.from ASC
+    WITH node_with_dup_edges, edge_type, edge_branch, peer, active_from, head(collect(deleted_edge.from)) AS deleted_from
+    CALL {
+        WITH node_with_dup_edges, edge_type, edge_branch, peer, active_from, deleted_from
+        MATCH (node_with_dup_edges)-[active_e {branch: edge_branch, status: "active"}]-(peer)
+        WHERE type(active_e) = edge_type
+        WITH node_with_dup_edges, edge_type, edge_branch, peer, active_from, deleted_from, active_e
+        ORDER BY elementId(active_e)
+        LIMIT 1
+        SET active_e.from = active_from
+        SET active_e.to = deleted_from
+        WITH node_with_dup_edges, edge_type, edge_branch, peer
+        MATCH (node_with_dup_edges)-[active_e {branch: edge_branch, status: "active"}]-(peer)
+        WHERE type(active_e) = edge_type
+        WITH active_e
+        ORDER BY elementId(active_e)
+        SKIP 1
+        DELETE active_e
+    }
+    CALL {
+        WITH node_with_dup_edges, edge_type, edge_branch, peer, deleted_from
+        MATCH (node_with_dup_edges)-[deleted_e {branch: edge_branch, status: "deleted"}]-(peer)
+        WHERE type(deleted_e) = edge_type
+        WITH node_with_dup_edges, edge_type, edge_branch, peer, deleted_from, deleted_e
+        ORDER BY elementId(deleted_e)
+        LIMIT 1
+        SET deleted_e.from = deleted_from
+        WITH node_with_dup_edges, edge_type, edge_branch, peer
+        MATCH (node_with_dup_edges)-[deleted_e {branch: edge_branch, status: "deleted"}]-(peer)
+        WHERE type(deleted_e) = edge_type
+        WITH deleted_e
+        ORDER BY elementId(deleted_e)
+        SKIP 1
+        DELETE deleted_e
+    }
+}
+"""
+
+# add missing IS_PART_OF edges
+
+query = """
+// nodes missing IS_PART_OF edges
+MATCH (n:Node)
+WHERE NOT exists((n)-[:IS_PART_OF]->(:Root))
+CALL {
+    WITH n
+    // get each branch this node has edges on
+    MATCH (n)-[e {status: "active"}]-()
+    // for each branch, return the earliest active edge
+    WITH DISTINCT n, active_e.branch AS branch
+    MATCH (n)-[active_e {status: "active", branch: branch}]-()
+    ORDER BY active_e.from ASC
+    RETURN head(collect(active_e)) AS earliest_active, branch
+}
+CALL {
+    WITH n
+    // get each branch this node has edges on
+    MATCH (n)-[attr_edge:HAS_ATTRIBUTE]->()
+    WITH DISTINCT n, attr_edge.branch AS branch
+
+    
+}
+"""
```

Comment:
It would be helpful to have some comments explaining the kind of expected corrupted state this query fixes. Also, we should have a test for it, that would ideally reproduce the expected corrupted state